packages/hardhat-solhint/src/index.ts

```diff
@@ -104,26 +104,19 @@ function printReport(reports: any) {
 
 subtask("hardhat-solhint:run-solhint", async (_, { config }) => {
   const { processPath } = require("solhint/lib/index");
-  return processPath(
-    /**
-     * Solhint passes the following path to the `node-glob` library.
-     * `path.join()` and `path.relative()` return back-slashes as path separator on Windows.
-     * However, glob pattern paths must use forward-slashes as path separators, since "\" is an escape character to match literal glob pattern characters.
-     * That's why we replace back-slashes with forward-slashes below.
-     *
-     * @see https://github.com/isaacs/node-glob/tree/v8.0.3#windows
-     *
-     * The files paths matching the glob pattern are then passed to `node-ignore`, which requires relative pathnames.
-     * For this reason, we pass a relative pathname to `processPath()`.
-     *
-     * @see https://github.com/kaelzhang/node-ignore/tree/5.2.4#1-pathname-should-be-a-pathrelatived-pathname
-     */
-    relative(
-      config.paths.root,
-      join(config.paths.sources, "**", "*.sol")
-    ).replace(/\\/g, "/"),
-    await getSolhintConfig(config.paths.root)
-  );
+
+  // Create a glob pattern that matches all the .sol files within the sources folder
+  const solFilesGlob = join(config.paths.sources, "**", "*.sol");
+
+  // Make glob pattern relative to the current working directory
+  // See https://github.com/kaelzhang/node-ignore/tree/5.2.4#1-pathname-should-be-a-pathrelatived-pathname
+  const relativeGlob = relative(process.cwd(), solFilesGlob);
```

Comment:
To be honest, I'm not 100% sure. First, I was using `config.paths.root` but I've changed it to `process.cwd()` in my last commit because it made more sense to me. [Solhint calls `fs.readFileSync(file)`](https://github.com/protofire/solhint/blob/master/lib/index.js#L55) where `file` is the relative path of the file to lint. I.e. in the context of Solhint, `process.cwd()` is the reference directory from which relative paths start.

In the tests, both `relative(config.paths.root, solFilesGlob)` and `relative(process.cwd(), solFilesGlob)` work. 
An edge case would be if Hardhat's `root` folder is different from the current working directory.
For instance:
```js
  paths: {
    root: "..",
    sources: "solhintignore-project/contracts",
  },
```
In this case, with paths relative to `process.cwd()`, the tests would pass, while with paths relative to `config.paths.root` the tests wouldn't pass.

But this is a really unusual case. In most cases, `config.paths.root` work fine.

If we decide to use `process.cwd()`, then I'll have to make sure that all the ignore globs are relative to the current working directory as well.

If we decide to ignore this edge case (hardhat.root ≠ cwd), then I'll be happy to replace `process.cwd()` with `config.paths.root`. 

I think I'm overthinking it 😅  Let me know what's best in your opinion.

Reply:
Hey @fvictorio! I've replaced `process.cwd()` with `config.paths.root` so as not to block this PR. It's highly unlikely that users will complain about the edge case I mentioned yesterday anyway.